pydicom/dataset.py

```diff
@@ -408,6 +412,73 @@ def __init__(self, *args: _DatasetType, **kwargs: Any) -> None:
 
         self.file_meta: FileMetaDataset
 
+    @property
+    def parent_seq(self) -> "Optional[weakref.ReferenceType[Sequence]]":
+        """Return a weak reference to the parent
+        :class:`~pydicom.sequence.Sequence`.
+
+        .. versionadded:: 2.4
+
+            Returned value is a weak reference to the parent ``Sequence``.
+        """
+        return self._parent_seq
+
+    @parent_seq.setter
+    def parent_seq(self, value: "Sequence") -> None:
+        """Set the parent :class:`~pydicom.sequence.Sequence`
+
+        .. versionadded:: 2.4
+        """
+        if value != self._parent_seq:
+            self._parent_seq = weakref.ref(value)
+
+    @property
+    def parent(self) -> "Optional[weakref.ReferenceType[Dataset]]":
+        """Return a weak reference to the parent Sequence's
+        parent Dataset.
+
+        .. deprecated:: 2.4
+        """
+        if config._use_future:
+            raise AttributeError("Future: Dataset.parent is removed in v3.x")
+        else:
+            warnings.warn(
+                "Dataset.parent will be removed in pydicom 3.0",
+                DeprecationWarning
+            )
+
+            parent_ref = self.parent_seq
+            if parent_ref is None:
+                return None
+
+            return cast("Sequence", parent_ref()).parent_dataset
+
+    @parent.setter
+    def parent(self, value: "Sequence") -> None:
+        """Set the parent :class:`~pydicom.sequence.Sequence`
+
+        .. deprecated:: 2.4
+        """
+        if config._use_future:
+            raise AttributeError("Future: Dataset.parent is removed in v3.x")
+        else:
+            warnings.warn(
+                "Dataset.parent will be removed in pydicom 3.0",
```

Comment:
This does not do anything - if somebody is indeed using this (which I doubt), this will break the code. I would add the setter code, or at least add a note to the warning that the code is non-functional

Reply:
Adding the actual setting code...